### PR TITLE
Feature/VP-3733: Show more then one users wish list

### PR DESCRIPTION
**Pull request by AliveMen**
### Problem
VP-3733: Wish list not shown when user create it in profile

### Solution
Add new GraphQL query to receive more than one with list 

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

File: VirtoCommerce.Storefront/Domain/Cart/CartService.cs [hunk 1]
```diff
@@ -130,6 +130,28 @@ public virtual async Task<IPagedList<ShoppingCart>> SearchCartsAsync(CartSearchC
...
         }
 
+        public virtual async Task<IPagedList<WishListDto>> SearchAccountListsAsync(CartSearchCriteria criteria)
+        {
+            if (criteria == null)
+            {
+                throw new ArgumentNullException(nameof(criteria));
+            }
+
+            var cacheKey = CacheKey.With(GetType(), "SearchCartsAsync", criteria.GetCacheKey());
```
Comment (yecli) [inline VirtoCommerce.Storefront/Domain/Cart/CartService.cs:140]: "SearchCartsAsync" should not be used. Please use nameof(<your method>)

File: VirtoCommerce.Storefront/Domain/Cart/CartService.cs [hunk 2]
```diff
@@ -130,6 +130,31 @@ public virtual async Task<IPagedList<ShoppingCart>> SearchCartsAsync(CartSearchC
             });
         }
 
+        public virtual async Task<IPagedList<CartDescriptionDto>> SearchAccountListsAsync(CartSearchCriteria criteria)
```
Comment (yecli) [inline VirtoCommerce.Storefront/Domain/Cart/CartService.cs:133]: We should use and change the existing `SearchCartsAsync` method, it should return response with multiple carts and totalCount.
PLease ensure proper CartType is used - in APIListController - wishlist, and default in ICartBuilder

File: VirtoCommerce.Storefront.Model/Cart/Services/ICartService.cs [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 using System.Threading.Tasks;
 using PagedList.Core;
+using VirtoCommerce.Storefront.Model.Contracts;
 
 namespace VirtoCommerce.Storefront.Model.Cart.Services
 {
     public interface ICartService
     {
         Task<IPagedList<ShoppingCart>> SearchCartsAsync(CartSearchCriteria criteria);
+        Task<IPagedList<CartDescriptionDto>> SearchAccountListsAsync(CartSearchCriteria criteria);
```
Comment (yecli) [inline VirtoCommerce.Storefront.Model/Cart/Services/ICartService.cs:11]: Remove this method, please. Need to use `SearchCartsAsync`, as we need not only description but also Items in the ApiListController

File: VirtoCommerce.Storefront/Controllers/Api/ApiListsController.cs [hunk 4]
```diff
@@ -141,7 +142,7 @@ public async Task<ActionResult> ClearList(string listName, string type)
         // POST: storefrontapi/lists/search
         [HttpPost("search")]
         [ValidateAntiForgeryToken]
-        public async Task<ActionResult<ShoppingCartSearchResult>> SearchLists([FromBody] CartSearchCriteria searchCriteria)
+        public async Task<ActionResult<IList<ShoppingCart>>> SearchLists([FromBody] CartSearchCriteria searchCriteria)
```
Comment (yecli) [inline VirtoCommerce.Storefront/Controllers/Api/ApiListsController.cs:145]: You should return ShoppingCartSearchResult, public contracts cannot not be broken.

File: VirtoCommerce.Storefront/Domain/Cart/CartService.cs [hunk 5]
```diff
@@ -120,31 +119,33 @@ public virtual async Task<IPagedList<ShoppingCart>> SearchCartsAsync(CartSearchC
...
             });
         }
 
-        private async Task<ShoppingCartDto> SearchCartAsync(CartSearchCriteria criteria)
+        private async Task<SearchCartResponseDto> InnerSearchCartsAsync(CartSearchCriteria criteria)
         {
-            var query = QueryHelper.GetCart(
-                storeId: criteria.StoreId,
-                cartName: criteria.Name,
+            var query = QueryHelper.SearchCart(storeId: criteria.StoreId,
```
Comment (yecli) [inline VirtoCommerce.Storefront/Domain/Cart/CartService.cs:139]: SearchCart**s**

File: VirtoCommerce.Storefront/Infrastructure/QueryHelper.cs [hunk 6]
```diff
@@ -11,12 +11,12 @@ public static string GetCart(string storeId, string cartName, string userId, str
             }}
         }}";
 
-        public static string GetWishLists(string storeId, string userId, string cultureName, string currencyCode, string type, string selectedFields = null)
+        public static string SearchCart(string storeId, string userId, string cultureName, string currencyCode, string type, string sort, int skip, int take, string selectedFields = null)
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply VirtoCommerce.Storefront/Infrastructure/QueryHelper.cs]: done

File: VirtoCommerce.Storefront/Controllers/Api/ApiListsController.cs [hunk 7]
```diff
@@ -141,7 +142,7 @@ public async Task<ActionResult> ClearList(string listName, string type)
         // POST: storefrontapi/lists/search
         [HttpPost("search")]
         [ValidateAntiForgeryToken]
-        public async Task<ActionResult<ShoppingCartSearchResult>> SearchLists([FromBody] CartSearchCriteria searchCriteria)
+        public async Task<ActionResult<IList<CartDescriptionDto>>> SearchLists([FromBody] CartSearchCriteria searchCriteria)
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply VirtoCommerce.Storefront/Controllers/Api/ApiListsController.cs]: reverted

File: VirtoCommerce.Storefront/Controllers/Api/ApiListsController.cs (hunk 4 above)
Comment (AliveMen) [reply VirtoCommerce.Storefront/Controllers/Api/ApiListsController.cs]: done

File: VirtoCommerce.Storefront/Domain/Cart/CartService.cs (hunk 5 above)
Comment (AliveMen) [reply VirtoCommerce.Storefront/Domain/Cart/CartService.cs]: done

**Pull request merged**